### ISSUE TITLE
Loss function when using custom eval_metric

### ISSUE THREAD

**Issue by Mrugankakarte** [dmlc/xgboost#3645]
Hi,
I'm trying to understand what is happening behind the scene. What is the loss function used by XGBoost when **objective = 'binary:logistic'**, **eval_metric=cost**, **scale_pos_weight=60**?

cost = 10\*false positives + 500\*false negatives 

Is the loss function 'merror'...If yes then how is the tree fitted on derivatives of residuals? or Is it logloss...if yes then how is the parameter 'scale_pos_weight' used?

**Comment by hcho3** [comment]
Since you are using objective='binary:logistic', the loss function used for training is the logistic loss. For the use of `scale_pos_weight`, see https://github.com/dmlc/xgboost/blob/v0.60/src/objective/regression_obj.cc.

**Comment by Mrugankakarte** [comment]
Thanks a lot for clarifying my doubt.
